source_pubsub/client.go

```diff
@@ -118,25 +130,28 @@ func NewGRPCClient(ctx context.Context, config Config, sdkPos sdk.Position) (*Pu
 		replayPreset = proto.ReplayPreset_EARLIEST
 	}
 
+	sdkPos.SetTopics(config.TopicNames)
```

Comment:
Wouldn't this overwrite and reset the replayID which were fetched from the original Position?

Reply:
The way I understand this, each startCDC method takes a topic and should resume from the topics replayID, so there isn't a single replay ID for all topics?

Reply:
Nope! So if lets say there were two topics in config on first run, and we save their position. If on a new run another topic is added, the parsed positions wouldn't include it so we want to set it replayID to nil 

func (p Topics) SetTopics(topics []string) {
	for _, topic := range topics {
		if _, ok := p.Topics[topic]; !ok {
			replayEvent := TopicPosition{
				ReplayID: nil,
			}
			p.Topics[topic] = replayEvent
		}
	}
}

Only  if the parsed position doesnt have a position for a topic is when we set it to nil

Reply:
OK, I see. Thanks